### PR TITLE
Prime factorization using sieve

### PR DESCRIPTION
**Pull request by kashika3101**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

A program to find out the prime factors of a number

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#437

### CODE REVIEW

File: C/ds/Primefactorization_Sieve.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	if (number > 2)                            //condition if n is a prime number greater than 2
+		printf ("%d ", number); 
+} 
+
+
+int main() 
+{ 
+	int number ;
+    scanf("%d",&number);
```
Comment (HarshCasper) [inline C/ds/Primefactorization_Sieve.c]: Indent the Code properly

File: C/ds/Primefactorization_Sieve.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    printf ("The prime factors of the number %d are :",number );
```
Comment (HarshCasper) [inline C/ds/Primefactorization_Sieve.c]: Make proper Console Statements

File: C/ds/Primefactorization_Sieve.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    while ((i*i) <= number)
+    {
+        if (primes[i] != 0)
+        {
+            for(j=2; j<number; j++)
+            {
+                if (primes[i]*j > number)
+                    break;
+                else
+                                                // marking the elements as 0
```
Comment (HarshCasper) [inline C/ds/Primefactorization_Sieve.c]: Remove extra Whitespaces

File: C/ds/Primefactorization_Sieve.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+   	while (number%2 == 0)                       // check if a number is even
+	{ 
+		printf("%d ", 2); 
+		number = number/2; 
+	} 
+
+
+	for (int i = 3; i <= sqrt(number); i = i+2) //the number must be odd 
+	{ 
+		                                        // While i divides n, print i and divide n
```
Comment (HarshCasper) [inline C/ds/Primefactorization_Sieve.c]: Use Multi-Line comments instead of Single-Line

File: C/ds/Primefactorization_Sieve.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+		{ 
+			printf("%d ", i); 
+			number = number/i; 
+		} 
+	} 
+
+	if (number > 2)                            //condition if n is a prime number greater than 2
+		printf ("%d ", number); 
+} 
+
```
Comment (HarshCasper) [inline C/ds/Primefactorization_Sieve.c]: Remove extra Lines

File: C/ds/Primefactorization_Sieve.c [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	{ 
+		                                        // While i divides n, print i and divide n 
+		while (number%i == 0) 
+		{ 
+			printf("%d ", i); 
+			number = number/i; 
+		} 
+	} 
+
+	if (number > 2)                            //condition if n is a prime number greater than 2
```
Comment (HarshCasper) [inline C/ds/Primefactorization_Sieve.c]: Use proper Comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Kindly resolve these reviews. It will be ready to merge then.

**Review by kashika3101 — COMMENTED** [review]
I have made the changes requested.

File: C/ds/Primefactorization_Sieve.c (hunk 1 above)
Comment (kashika3101) [reply C/ds/Primefactorization_Sieve.c]: sure.

File: C/ds/Primefactorization_Sieve.c (hunk 4 above)
Comment (kashika3101) [reply C/ds/Primefactorization_Sieve.c]: resolved the issue. Kindly review it.

**Comment by akanksha-saini** [comment]
I would like to provide the solution for "Prime Sieve" in C++  @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99 @siddharth25pandey

**Comment by SKAUL05** [comment]
> I would like to provide the solution for "Prime Sieve" in C++ @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99 @siddharth25pandey



> I would like to provide the solution for "Prime Sieve" in C++ @SKAUL05 @Kajol-Kumari @Abhijit2505 @plazzy99 @siddharth25pandey

@akanksha-saini @kashika3101  is still resolving issues related to current PR. I would request you to keep an eye on this PR if it's not merged and closed then u can provide your solution.
Meanwhile, head over to the Issues section to know about currently open issues.

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
edit readme

**Comment by kashika3101** [comment]
@plazzy99 I have updated the Readme.
@HarshCasper I have resolved all the issue as per recommendations.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@kashika3101 add space and time complexity

**Comment by kashika3101** [comment]
@iamrajiv @plazzy99 @HarshCasper please review

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@kashika3101 move this program to `math` section, not in `ds` section

**Review by iamrajiv — APPROVED** [review]
@kashika3101 LGTM!

**Comment by vatsalkesarwani12** [comment]
resolve merge conflict

**Review by atarax665 — APPROVED** [review]
Remove merge conflict, else everything looks fine. :+1:

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
Optimize primeFactor function

**Pull request closed without merging**